aips/aip-resource-access.md

```diff
@@ -0,0 +1,196 @@
+- Since looking at code is the only way to reason about effects, _calling unknown_ code (dynamic dispatch across trust boundaries) is unsafe, and therefore currently forbidden in Move.
+
+## Impact
+
+Significant impact is expected in the medium term on two frontiers: on the one hand sharding and parallelization, on the other enabling other new language features like dynamic dispatch. 
+
+
+## Alternative solutions
+
+In Solana, a transaction must declare upfront the accounts it reads or writes from. Similarily, in Sui Move a transaction declares what objects it accesses. Using Aptos Objects as basis for what is proposed here has been discussed. Arguably, the approach described here is closer to the core Move language, as it does not depend on library extensions like objects. The approach is also more general, as it allows a variety of different access patterns, including negation.
+
+
+## Specification
+
+### Source Language
+
+The existing `acquires T` notation in Move is replaced by an extended syntax:
+
+```move
+fun f() acquires R {}                      // today: reads or write at any address; only local to module
+fun f() reads M1::R writes M2::T {}        // new access kinds -- note reference to types from other modules
+fun f<T> acquires R<T> {}                  // type instantiations supported
+fun f() acquires 0x42::*                   // wildcards
+fun f(x: T) acquires *(address_of(x))      // data dependency
+fun f() acquires !0x1::* {}                // negation
+```
+
+### File Format
+
+The existing `FunctionHandle` is extended by a field which contains an optional list of access specifiers. Each access specifier consists of a kind (read/write/..), a resource specifier, and an address specifier. Both those data types are chosen to represent the various forms of patterns as seen in the source language. For details, see the implementation.
+
+### Semantics
+
+#### Access Specifier Subsumption
+
+The conceptual syntax of access specifiers is as follows:
+
+```
+AccessSpecifier := Kind ResourceSpecifier AddressSpecifier
+ResourceSpecifier := * | Address::* | Address::Module::* | Address::Module::Resource [ TypeArgs ]
+AddressSpecifier := * | Address | Parameter | Function Parameter
+Kind := acquires | reads | writes
+```
+
+The basic function specifying the semantics of access specifiers is subsumption. One access specifier subsumes another one if all access allowed in the other are also allowed in the subsuming one. For `ResoureSpecifier` this looks as below. Notice that transitivity of subsumption is assumed:
+
+```
+*         >=     _
+A::*      >=     A::M::*
+A::M::*   >=     A::M::R
+A::M::R   >=     A::M::R<TS>
+```
+
+Subsumption for `AddressSpecifier` is similar but depends on an environment to obtain values for parameters and specific functions for them. Assume this environment is called `E`:
+
+```
+*         >=     _
```

Comment:
I guess (from knowledge of Rust) that you mean * >>= anything, but I think you should be more explicit here.

Reply:
Done